### DISCUSSION TITLE
Is there a way of generating the deprecated stuff?

### DISCUSSION THREAD

**Discussion by luismanfroni** [dotansimha/graphql-code-generator#8682] [Q&A]
I have a deprecated mutation using "deprecation_reason" but isn't generating any metadata of it, is there a way of activating that via any flag or config?

**Comment by saihaj** [comment]
Can you share a  [minimal reproduction](https://en.wikipedia.org/wiki/Minimal_working_example) for this. ASFAIK we include `deprecated` fields by default. You can either use [StackBlitz](https://stackblitz.com/) or [CodeSandbox](https://codesandbox.io/) or a link to public GitHub repository for providing your reproduction.

**Reply by MrTact** [reply]
ChatGPT certainly seems to think so:

![image (1)](https://user-images.githubusercontent.com/1398633/234702016-63373364-4f28-403f-af4b-9c59ed1faf37.png)

EDIT: For future reference, in case anyone trips over this post, `skipDeprecated` is NOT a config option for either graphql-codegen or the typescript plugin! This is ChatGPT getting high on its own supply.  **This WILL NOT work**.

In non-LOL news, I am also running into this problem. I don't have a minimal repro yet, but I can mention this is an Input type. I poked around the Input generation code & couldn't quite home in it yet.

If I can generate a minimal case, I will open a bug on this.

**Reply by MrTact** [reply]
Okay so it looks like it's not a problem with graphql-codegen at all. In our case, it appears the server lib we are using isn't returning the deprecated fields on introspection, so that may be OP's problem as well.
